src/bsp/platform/cc3200/xi_bsp_io_fs_cc3200.c

```diff
@@ -100,9 +113,14 @@ xi_state_t xi_bsp_io_fs_open( const char* const resource_name,
 
         _i32 file_handle = 0;
 
-        if ( 0 != sl_FsOpen( ( _u8* )resource_name,
-                             xi_bsp_io_fs_open_flags_to_sl_flags( size, open_flags ),
-                             NULL, &file_handle ) )
+        /* prevent accidental wirte of CC3200 firmware files by limiting access rights to
+         * read only */
+        const _u32 access_mode =
+            ( 1 == xi_bsp_io_fs_is_this_cc3200_firmware_filename( resource_name ) )
+                ? FS_MODE_OPEN_READ
+                : xi_bsp_io_fs_open_flags_to_sl_flags( size, open_flags );
```

Comment:
Shouldn't this be an error instead of changing the intended open mode without reporting that you've done so?